core/src/main/java/net/sourceforge/jnlp/util/WindowsDesktopEntry.java

```diff
@@ -204,4 +207,36 @@ public File getLinuxMenuIconFile() {
     private String quoted(URL url) {
         return DOUBLE_QUOTE + url.toExternalForm() + DOUBLE_QUOTE;
     }
+    
+	private final static String desktopPath() {
+		final char QUOT = '"';
+		final String registryEntry = "HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Shell Folders";
+		final String desktopProperty = "Desktop";
+
+		final ProcessBuilder pb = new ProcessBuilder("reg", "query", QUOT + registryEntry + QUOT, "/v",
+				QUOT + desktopProperty + QUOT);
+
+		try {
+			final Process p = pb.start();
+			p.waitFor(5, TimeUnit.SECONDS);
+			if (p.exitValue() == 0) {
+				try (final InputStream is = p.getInputStream()) {
+					final String output = IOUtils.readContentAsUtf8String(is);
+					if (output != null) {
+						final String typeName = "REG_SZ";
+						final int typeIndex = output.indexOf(typeName);
+						if (typeIndex != -1) {
+							final String desktopPath = output.substring(typeIndex + typeName.length());
+							if (desktopPath != null) {
+								return desktopPath.trim();
+							}
+						}
+					}
+				}
+			}
+		} catch (final Exception e) {
+			LOG.error("Could not retrieve path to client desktop folder from registry", e);
+		}
+		return System.getenv("userprofile") + "/Desktop";
```

Comment:
Instead of using this trivial implementation which is known to fail could you implement the fallback using the method proposed in the issue https://github.com/karakun/OpenWebStart/issues/243

```
FileSystemView.getFileSystemView().getHomeDirectory().getAbsolutePath()
```